dove/tests/cmd_dove_clean.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg(test)]
+
+use std::path::{Path, PathBuf};
+use std::fs::{remove_dir_all, read_to_string};
+use fs_extra::file::write_all;
+use dove::cli::execute;
+
+/// $ cargo run -- clean
+/// $ dove clean
+/// project name: demoproject_16
+#[test]
+fn clean() {
+    // Path to dove folder
+    let dove_folder = {
+        let mut folder = Path::new(".").canonicalize().unwrap();
+        if folder.to_str().unwrap().find("dove").is_none() {
+            folder.push("dove");
+        }
+        folder
+    };
+    // Project name and path
+    let project_name = "demoproject_16";
+    let project_folder = {
+        let mut folder = dove_folder.clone();
+        folder.push(project_name);
+        if folder.exists() {
+            remove_dir_all(&folder).expect(&format!(
+                "[ERROR] Couldn't delete project directory: {}",
+                folder.to_str().unwrap()
+            ));
+        }
+        folder
+    };
+    // $ cargo run -- new demoproject_16 -d pont
+    // $ dove new demoproject_16 -d pont
+    {
+        let args = &["dove", "new", project_name];
+        let command_string: String = args.join(" ").to_string();
+        execute(args, dove_folder.clone()).expect(&format!("[COMMAND] {}", &command_string));
+        set_dependencies_local_move_stdlib(&project_folder);
+    };
+    // $ cargo run -- build
+    // $ dove build
+    {
+        let args = &["dove", "build"];
+        let command_string: String = args.join(" ").to_string();
+        execute(args, project_folder.clone()).expect(&format!(
+            "[COMMAND] {}\r\n[FOLDER] {}",
+            &command_string,
+            project_folder.to_str().unwrap()
+        ));
+    }
+
+    // $ cargo run -- clean
+    // $ dove clean
+    {
+        let args = &["dove", "clean"];
+        let command_string: String = args.join(" ").to_string();
+        execute(args, project_folder.clone()).expect(&format!(
+            "[COMMAND] {}\r\n[FOLDER] {}",
+            &command_string,
+            project_folder.to_str().unwrap()
+        ));
+
+        let mut project_target = project_folder.clone();
+        project_target.push("target");
+
+        assert!(
+            !project_target.exists(),
+            "Directory was not deleted: {}",
+            project_target.to_str().unwrap_or(" - ")
```

Comment:
nit: I believe `.display()` method can be used here. It's probably the best way to print `Path`.